src/cascadia/UnitTests_Control/ControlInteractivityTests.cpp

```diff
@@ -846,11 +852,99 @@ namespace ControlUnitTests
+        const til::point terminalPosition0{ 5, 5 };
+        const til::point cursorPosition0{ terminalPosition0 * fontSize };
+        interactivity->PointerPressed(leftMouseDown,
+                                      WM_LBUTTONDOWN, //pointerUpdateKind
+                                      0, // timestamp
+                                      modifiers,
+                                      cursorPosition0.to_core_point());
+
+        // These first two bits are a test for GH#11290
+        Log::Comment(L"Scroll up, click the terminal, the event should be clamped to y=0");
+        core->UserScrollViewport(0);
+
+        // Viewport is now on row 0, but the mutable viewport is way below that.
+        // Mouse even should still come in clamped to the real viewport.
+
+        expectedOutput.push_back(L"\x1b[M &&");
+        interactivity->PointerPressed(leftMouseDown,
+                                      WM_LBUTTONDOWN, //pointerUpdateKind
+                                      0, // timestamp
+                                      modifiers,
+                                      cursorPosition0.to_core_point());
+
+        Log::Comment(L"Click on the terminal outside the mutable viewport, see that it's clamped to the viewport");
+        const til::point terminalPosition1{ originalViewport.Width() + 5, 5 };
+        const til::point cursorPosition1{ terminalPosition1 * fontSize };
+
+        // The viewport is only 30 wide, so clamping 35 to the buffer size gets
+        // us 29, whech converted is (32 + 29 + 1) = 62 = '>'
+        expectedOutput.push_back(L"\x1b[M >&");
+        interactivity->PointerPressed(leftMouseDown,
+                                      WM_LBUTTONDOWN, //pointerUpdateKind
+                                      0, // timestamp
+                                      modifiers,
+                                      cursorPosition1.to_core_point());
+
+        Log::Comment(L"Switch to alt buffer");
+        term.Write(L"\x1b[?1049h");
+
+        Log::Comment(L"Click on a spot that's still outside the buffer");
+        expectedOutput.push_back(L"\x1b[M >&");
+        interactivity->PointerPressed(leftMouseDown,
+                                      WM_LBUTTONDOWN, //pointerUpdateKind
+                                      0, // timestamp
+                                      modifiers,
+                                      cursorPosition1.to_core_point());
+
+        Log::Comment(L"Resize the terminal to be 10 columns wider");
+        core->SizeChanged(40, 20);
+
+        Log::Comment(L"Click on a spot that's NOW INSIDE the buffer");
+        expectedOutput.push_back(L"\x1b[M C&");
+        interactivity->PointerPressed(leftMouseDown,
+                                      WM_LBUTTONDOWN, //pointerUpdateKind
+                                      0, // timestamp
+                                      modifiers,
+                                      cursorPosition1.to_core_point());

```

Comment:
Callback-based tests scare me a little bit, because it is so easy to author one that *fires no input* (due to the bug under test) and the callback never fires to validate it. That's why I added "verifyExpectedInputDrained" to one of the test fixtures...
Do we need a guarantee like that here? How do we know that PointerPressed emitted anything?